core/src/main/java/ai/timefold/solver/core/impl/solver/ConsumerSupport.java

```diff
@@ -62,6 +65,11 @@ void consumeFirstInitializedSolution(Solution_ firstInitializedSolution) {
         scheduleFirstInitializedSolutionConsumption();
     }
 
+    // Called on the solver thread when it starts
+    void triggerStartSolverJob() {
```

Comment:
Shouldn't this run on a different thread? 
The solver thread cannot be blocked by user code. Have you seen what platform does in the listeners? It's heavy.